clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -941,9 +943,11 @@ class KernelObjVisitor {
       else if (FieldTy->isArrayType()) {
         if (KF_FOR_EACH(handleArrayType, Field, FieldTy))
           VisitArrayElements(Field, FieldTy, handlers...);
-      } else if (FieldTy->isScalarType() || FieldTy->isVectorType())
+      } else if (FieldTy->isScalarType() || FieldTy->isVectorType()) {
         KF_FOR_EACH(handleScalarType, Field, FieldTy);
-      else
+      } else if (FieldTy->isUnionType()) {
+        KF_FOR_EACH(handleUnionType, Field, FieldTy);
+      } else
```

Comment:
If code of "handleScalarType" in "IntHeaderCreator" also suits for unions, we could unify it across scalars, unions and vectors, i.e. here you will have something like
```
} else if (FieldTy->isScalarType() || FieldTy->isVectorType()
 || FieldTy->isUnionType()) {
KF_FOR_EACH(handleSimpleType, Field, FieldTy);
}
```
i.e. `handleScalarType` can be renamed to something more generic (`handleSimpleType` in my example) and used for scalars, vectors and unions..

Reply:
@Fznamznon, It seems like"handleScalarType" suits for unions. So this is renamed to "handleSimpleType".